modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -277,6 +279,11 @@ else if (Objects.equals(
 						engine,
 						ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY)) {
 
+				if (StringUtil.count(script, StringPool.NEW_LINE) > 2987) {
+					throw new ObjectValidationRuleScriptException(
+						"the-maximum-number-of-lines-available-is-2987");
+				}
+
```

Comment:
@danvb112 can you put the catch like this?

```
catch (Exception exception) {
     if (_log.isDebugEnabled()) {
        _log.debug(exception);
    }

    if(exception instanceof ObjectValidationRuleScriptException){
        throw exception;
    }

    throw new ObjectValidationRuleScriptException("syntax-error");
}
```

could you test with the API call and check if the right exception is thrown? Thx.